### PR TITLE
Combobox: multi-select does not clear index on selection. Selected and checked properly supported

### LINKED ISSUE #11198: [Accessibility][Keyboard navigation]: User have to re-navigate the checkbox options to uncheck/check them at the same time.
Split out of issue #6920

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #11198
- [x] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/11397)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -86,6 +86,9 @@ interface IComboBoxOptionWrapperProps extends IComboBoxOption {
   /** True if the option is currently selected */
   isSelected: boolean;
 
+  /** True if the option is currently checked (multi-select) */
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: >checked (multi-select) [](start = 38, length = 22)

What about the single selected checked case?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -934,7 +937,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     }
 
     // clear all of the pending info
-    this._clearPendingInfo();
+    !this.props.multiSelect && this._clearPendingInfo();
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: >multiSelect [](start = 16, length = 11)

The change on this line looks a little scary... we'll need to make sure to validate this well

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: (and below)

---
In reply to: [355044289](https://github.com/OfficeDev/office-ui-fabric-react/pull/11397#discussion_r355044289) [](ancestors = 355044289)

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: maybe we need an additional state to track the currently focused control

File: packages/office-ui-fabric-react/src/components/__snapshots__/ComboBox.Basic.Example.tsx.shot [hunk 3]
```diff
@@ -3505,7 +3496,6 @@ exports[`Component Examples renders ComboBox.Basic.Example.tsx correctly 1`] = `
                           }
                       data-automationid="splitbuttonprimary"
                     >
-
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/__snapshots__/ComboBox.Basic.Example.tsx.shot]: Looks like there's only whitespace changes in this file, is that expected?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (joschect) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Single select doesn't have a checked case I thought. Isn't multiselect the only one that cares about checked?

**Pull request merged**